src/adapter/src/catalog/state.rs

```diff
@@ -123,6 +123,7 @@ pub struct CatalogState {
     pub(super) default_privileges: DefaultPrivileges,
     pub(super) system_privileges: PrivilegeMap,
     pub(super) comments: CommentsMap,
+    pub(super) source_references: BTreeMap<GlobalId, SourceReferences>,
```

Comment:
It feels like this may be a bit cleaner if we store it directly in the `Source` object as a field instead of in a separate map. Then we wouldn't have to worry about the state of `source_references` and `entry_by_id` diverging. Also you wouldn't have to add new handling in `apply.rs`.

Reply:
I've done this refactoring, but now I'm actually concerned this may not be the best design since I'm realizing it forces me to need to store the 'available_source_references' on the `CREATE SOURCE` statement objects in the catalog too, such that they can be populated on the `Source` created as the result of planning that statement.

This would make things more complicated since then I'd need to make sure the references stored on the statement are synchronized with the references stored in the durable catalog `source_reference` type, which seems like unnecessary denormalization.

Does that make sense to you? I think if I revert back to the original design it allows keeping these things entirely separate, and will make it easier to add the `REFRESH AVAILABLE SOURCES` statement which wont have to figure out how to alter the `CREATE SOURCE` statements too

Reply:
Okay I've refactored a little further and got this working with some WIP changes in a follow-on PR. It is a little awkward since we are storing the source-references inside the durable catalog as a distinct type, but in the in-memory catalog are storing them on the `Source` object, which means that we need to update the built-in system tables from two places.
However I was able to refactor things such that we don't need to store anything on the `CREATE SOURCE` statements such that the data is not duplicated in the durable catalog.

Let me know what you think!

Reply:
I see what you mean, I think both approaches have their pros and cons and I'll leave it up to you which you prefer. They're both OK with me. If you store them separately though, please add a catalog consistency check to `consistency.rs` that asserts the `GlobalId` references are valid.

Reply:
Great - I think the current approach is fine for now so I will merge this PR as-is. If it ends up being difficult to implement the `REFRESH AVAILABLE SOURCES` statement because of this approach I will refactor it at that point, and since that would only change the in-memory and not durable representation we shouldn't need to worry about compatibility